sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/timeExpressions.scala

```diff
@@ -290,3 +290,45 @@ object HourExpressionBuilder extends ExpressionBuilder {
     }
   }
 }
+
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(hour, minute, second) - Create time from hour, minute and second fields. For invalid inputs it will throw an error.",
+  arguments = """
+    Arguments:
+      * hour - the hour to represent, from 0 to 23
+      * minute - the minute to represent, from 0 to 59
+      * second - the second to represent, from 0 to 59.999999
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(6, 30, 45.887);
+       06:30:45.887
+      > SELECT _FUNC_(NULL, 30, 0);
+       NULL
+  """,
+  group = "datetime_funcs",
+  since = "4.1.0")
+// scalastyle:on line.size.limit
+case class MakeTime(
+    hours: Expression,
+    minutes: Expression,
+    secsAndMicros: Expression)
+  extends RuntimeReplaceable with ExpectsInputTypes {
```

Comment:
How about to use the implicit cast otherwise users have to cast seconds to the expected decimal:
```sql
spark-sql (default)> select make_time(6, 30, 45.887);
[DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE] Cannot resolve "make_time(6, 30, 45.887)" due to data type mismatch: The third parameter requires the "DECIMAL(16,6)" type, however "45.887" has the type "DECIMAL(5,3)". SQLSTATE: 42K09; line 1 pos 7;
'Project [unresolvedalias(make_time(6, 30, 45.887))]
+- OneRowRelation

spark-sql (default)> select make_time(6, 30, cast(45.887 as decimal(16,6)));
23445887000
```

Reply:
Fixed:
```scala
scala> spark.sql("select make_time(6, 30, 45.887)").first
val res0: org.apache.spark.sql.Row = [06:30:45.887]
```